tools/document-pii/src/item_collector.rs

```diff
@@ -0,0 +1,325 @@
+//! Collecting full paths to items and AST-nodes of types
+//!
+//! This module will iterate over all the rust files given and collect all of the full path names
+//! and the actual AST-node for the types defined in those paths. This is later needed for finding
+//! PII fields recursively.
+
+use std::collections::{BTreeMap, BTreeSet, HashMap};
+use std::fs::{self, DirEntry};
+use std::io::BufRead;
+use std::path::{Path, PathBuf};
+
+use anyhow::anyhow;
+use syn::punctuated::Punctuated;
+use syn::visit::Visit;
+use syn::{ItemEnum, ItemStruct, UseTree};
+
+use crate::EnumOrStruct;
+
+pub struct TypesAndScopedPaths {
+    // Maps the path name of an item to its actual AST node.
+    pub all_types: HashMap<String, EnumOrStruct>,
+    // Maps the paths in scope to different modules. For use in constructing the full path
+    // of an item from its type name.
+    pub scoped_paths: BTreeMap<String, BTreeSet<String>>,
+}
+
+/// The types and use statements items collected from the rust files.
+#[derive(Default)]
+pub struct AstItemCollector {
+    module_path: String,
+    /// Maps from the full path of a type to its AST node.
+    all_types: HashMap<String, EnumOrStruct>,
+    /// Maps from a module_path to all the types that are in the module's scope.
+    scoped_paths: BTreeMap<String, BTreeSet<String>>,
+}
+
+impl AstItemCollector {
+    fn insert_scoped_paths(&mut self, use_statements: Vec<String>) {
+        self.scoped_paths
+            .entry(self.module_path.clone())
+            .or_default()
+            .extend(use_statements);
+    }
+
+    /// Gets both a mapping of the full type to a type and its actual AST node, and also the
+    /// use_statements in its module, which is needed to fetch the types that it referes to in its
+    /// fields.
+    pub fn get_types_and_scoped_paths(paths: &[PathBuf]) -> anyhow::Result<TypesAndScopedPaths> {
```

Comment:
```suggestion
    pub fn collect(paths: &[PathBuf]) -> anyhow::Result<TypesAndScopedPaths> {
```

Since the struct name and doc already describes what it collects, we do not have to put this in to the name of the function. Simpler names will work much better here.